spec/enumerables_spec.rb

```diff
@@ -0,0 +1,148 @@
+require './lib/enumerables'
+
+RSpec.describe Enumerable do
+    let(:arr) { [1, 2, 3, 4, 3] }
+    let(:new_arr) { [ ] }
+    let(:arg) { 23 }
+    let(:count) { 5 }
+    let(:word) {['ChristLord', 'Living', 'Life', 'Love']}
+
+
+    describe '#.my_each' do
+
+        context 'When array and block are given'do
+            it 'returns the items in the array' do
+                expect( arr.my_each_with_index { |i, v| i}).to eql( arr.each_with_index { |i, v| i} )
+            end
+        end
+
+        context 'When no block is given' do
+            it 'returns an enumerator' do
+                expect(arr.my_each).to be_instance_of(Enumerator)
+            end
+        end
+    end
+
+    describe '#.my_each_with_index' do
+        context 'When no block is given' do
+            it 'returns an enumerator' do
+                expect(arr.my_each_with_index).to be_instance_of(Enumerator)
+            end
+        end
+    end
+
+    describe '#.my_select' do
+
+        context 'When array and block are given'do
+            it 'returns the selected items in the array' do
+                 expect( arr.my_select { |i| i > 3 }).to eql(arr.select { |i| i > 3})
+            end
+        end
+        context 'When no block is given' do
+            it 'returns an enumerator' do
+                 expect( arr.my_select {|i| i == 3 }).to eql( [ 3, 3 ] )
+                 expect(arr.my_select {|i| i == 3 }).to eql(arr.my_select {|i| i == 3 })
+            end
+        end
+     end
+
+    describe '#.my_all?' do
+
+        context 'When array and block are given'do
+
+            it 'returns false when not all the items in the array are true' do
+                expect( arr.my_all? { |i| i.even? }).to eql(false)
+            end
+
+            it 'returns an true or false depending on the argument' do
+                expect( arr.my_all?(/r/)).to eql(arr.all?(/r/))
+            end
+        end
+        context 'When no block is given' do
+            ints = [3,3,3]
+            it 'returns an true or false depending on the argument' do
+                expect( ints.my_all?(3)).to eql(ints.all?(3))
+            end
+        end
+    end
+
+    describe '#my_any?' do
+        context 'When a block is provided' do
+            it 'returns true when any of the number in the array is an integer' do
+                expect(arr.my_any? { |i| i.class == Integer }).to eql(true)
+            end
+
+            it 'returns true when any of the number in the array is true' do
+                expect( arr.my_any?(/we/)).to eq(arr.any?(/we/))
+            end
+        end
+
+        context 'When no block is given but arguments' do
+
+            it 'returns true when the class of one of the items in the array is similar to the arguments' do
+                expect(arr.my_any?(Integer)).to eql(true)
+            end
+        end
```

Comment:
You can make the message more readable and be specific to the task. like 'returns true when any of the items matches with the given class '